include/socgraph.php

```diff
@@ -190,181 +187,166 @@ function poco_load_worker($cid, $uid, $zcid, $url) {
 	);
 
 }
+/**
+ * @brief Sanitize the given gcontact data
+ *
+ * @param array $gcontact array with gcontact data
+ *
+ * Generation:
+ *  0: No definition
+ *  1: Profiles on this server
+ *  2: Contacts of profiles on this server
+ *  3: Contacts of contacts of profiles on this server
+ *  4: ...
+ *
+ */
+function sanitized_gcontact(&$gcontact) {
```

Comment:
By now I'm too dump for exceptions. I call it "sanitized_gcontact" since it always returns a boolean value, so it is readable like "if sanitized_gcontact then" ...

Is calling by reference a bad thing?